artifact/gem5art/artifact/_artifactdb.py

```diff
@@ -198,12 +200,83 @@ def searchByLikeNameType(self, name: str, typ: str, limit: int) -> Iterable[Dict
         for d in data:
             yield d
 
+class ArtifactFileDB(ArtifactDB):
+    """
+    This is a file-based database where Artifacts (as defined in artifacts.py)
+    are stored in a JSON file. This database does not keep track of files.
+
+    This database stores the data in three collections:
+    - artifacts: This stores the json serialized Artifact class
+    """
+
+    def __init__(self, uri :str) -> None:
+        """Initialize the mongodb connection and grab pointers to the databases
+           uri is the location of the database in a mongodb compatible form.
+           http://dochub.mongodb.org/core/connections.
+        """
+        self.db = FileDB(uri)
+
+    def put(self, key: UUID, artifact: Dict[str,Union[str,UUID]]) -> None:
+        """Insert the artifact into the database with the key"""
+        assert artifact['_id'] == key
+        assert isinstance(artifact['hash'], str)
+        self.db.insert_artifact(key, artifact['hash'], artifact)
+
+    def upload(self, key: UUID, path: Path) -> None:
+        """Upload the file at path to the database with _id of key"""
+        #with open(path, 'rb') as f:
+        #    self.fs.upload_from_stream_with_id(key, str(path), f)
+        self.db.register_file(key, path)
+
+    def __contains__(self, key: Union[UUID, str]) -> bool:
+        """Key can be a UUID or a string. Returns true if item in DB"""
+        if isinstance(key, UUID):
+            return self.db.has_uuid(key)
+        return self.db.has_hash(key)
+
+    def get(self, key: Union[UUID,str]) -> Dict[str,str]:
+        """Key can be a UUID or a string. Returns a dictionary to construct
+        an artifact.
+        """
+        artifact: List[Dict[str, str]] = []
+        if isinstance(key, UUID):
+            artifact = list(self.db.get_artifact_by_uuid(key))
+        else:
+            # This is a hash.
+            artifact = list(self.db.get_artifact_by_hash(key))
+        return artifact[0]
+
+    def downloadFile(self, key: UUID, path: Path) -> None:
+        """Do nothing."""
+        pass
```

Comment:
I would either assert that the file path in the "database" us the same as path, or call `cp`

Reply:
ping